src/store.ts

```diff
@@ -1,54 +1,23 @@
-import {Action, Reducer, StoreCreator} from './interfaces';
-import {combineReducers} from './utils'
-
 import {Observable} from 'rxjs/Observable';
-import {Subscription} from 'rxjs/Subscription';
 import {BehaviorSubject} from 'rxjs/subject/BehaviorSubject';
-import {Subject} from 'rxjs/Subject';
-import {Operator} from 'rxjs/Operator';
-
-//store specific operators
 import 'rxjs/add/operator/map';
 import 'rxjs/add/operator/distinctUntilChanged';
-import 'rxjs/add/operator/scan';
-import 'rxjs/add/operator/do';
-
-
-//public injectable dispatcher
-export class Dispatcher extends Subject<any> {
-  dispatch(action) {
-    this.next(action);
-  }
-}
-
-const createActionStream = dispatcher => dispatcher;
-
-
 
+import {Action, Reducer} from './interfaces';
+import {StoreBackend} from './store-backend';
+import {Dispatcher} from './dispatcher';
 
 export class Store<T> extends BehaviorSubject<T> {
-  private _sub: Subscription;
-  private _rootReducer: Reducer<any>;
-  private _reducers: any;
-  private _dispatcher: Dispatcher;
-  constructor(_dispatcher: Dispatcher, _reducers: any, initialState = {}) {
-    const rootReducer = combineReducers(_reducers);
-    super(rootReducer(initialState, { type: '@@ngrx/INIT' }));
-    this._rootReducer = rootReducer;
-    this._reducers = _reducers;
-    this._dispatcher = _dispatcher;
-    this.initialize(this.value);
-  }
+  _reducer: Reducer<any>
 
-  initialize(initialState) {
-    if (this._sub) {
-      this._sub.unsubscribe();
-    }
-    this._sub = this._dispatcher.scan(this._rootReducer, initialState).subscribe(state => super.next(state));
-  }
+  constructor(
+    private _dispatcher: Dispatcher<Action>,
+    backend: StoreBackend,
+    initialState?: T
+  ) {
+    super(initialState);
 
-  next(action: any) {
```

Comment:
next should be left in here, and now you've reintroduced the dispatcher, redirected through that. Store should still implement the `Observer` interface, and w/o overriding next, it would simply set the state and short circuit the whole process.